docs/user/exercises/programming-exercise-setup.rst

```diff
@@ -0,0 +1,110 @@
+Exercise Setup
+==============
+
+Exercise Creation
+-----------------
+
+1. **Open Course Management**
+
+- Open **Course Management**
+- Navigate into **Exercises** of your preferred course
+
+    .. figure:: programming/course-management-course-dashboard.png
+              :align: center
+
+2. **Generate programming exercise**
+
+- Click on **Generate new programming exercise**
+
+    .. figure:: programming/course-management-exercise-dashboard.png
+              :align: center
+
+- Fill out all required values and click on **Generate**
+
+    .. figure:: programming/create-programming-1.png
+              :align: center
+
+    .. figure:: programming/create-programming-2.png
+              :align: center
+
+  Result: **Programming Exercise**
+
+    .. figure:: programming/course-dashboard-exercise-programming.png
+              :align: center
+
+  Artemis creates the repositories:
+
+  - **Template:** template code, can be empty, all students receive this code at the beginning of the exercises
+  - **Test:** contains all test cases, e.g. based on JUnit, hidden for students
+  - **Solution:** solution code, typically hidden for students, can be made available after the exercise
+
+  Artemis creates two build plans
+
+  - **Template:** also called BASE, basic configuration for the test + template repository, used to create student build plans
+  - **Solution:** also called SOLUTION, configuration for the test + solution repository, used to manage test cases and to verify the exercise configuration
+
+  .. figure:: programming/programming-view-1.png
+            :align: center
+  .. figure:: programming/programming-view-2.png
+            :align: center
+  .. figure:: programming/programming-view-2.png
+            :align: center
+
+3. **Update exercise code in repositories**
+
+- **Alternative 1:** Clone the 3 repositories and adapt the code on your local computer in your preferred development environment (e.g. Eclipse)
+
+  - To execute tests, copy the template (or solution) code into a folder **assignment** in the test repository and execute the tests (e.g. using maven clean test)
+  - Commit and push your changes
+
+- **Alternative 2:** Open Edit in Editor in Artemis (in the browser) and adapt the code in online code editor
+
+  - You can change between the different repos and submit the code when needed
+
+- **Alternative 3:** Use IntelliJ with the Orion plugin and change the code directly in IntelliJ
+
+  **Edit in Editor**
+
+  .. figure:: programming/instructor-editor.png
+            :align: center
+
+- Check the results of the template and the solution build plan
+- They should not have the status **build failed**
```

Comment:
Here again, an inline image of the build failed status would really make it clear. There is a screenshot already of this in the exam mode students guide.